powersimdata/input/input_data.py

```diff
@@ -85,6 +87,33 @@ def get_data(self, scenario_info, field_name):
         _cache.put(key, data)
         return data
 
+    def get_profile_version(self, scenario_info, kind):
+        """Returns available raw profile either from server or local directory.
+
+        :param dict scenario_info: scenario information.
+        :param str kind: *'demand'*, *'hydro'*, *'solar'* or *'wind'*.
+        :return: (*list*) -- available profile version.
+        :raises ValueError: if kind not one of *'demand'*, *'hydro'*, *'solar'* or
+            *'wind'*.
+        """
+        if kind not in self.profile:
+            raise ValueError("kind must be one of " % " | ".join(self.profile))
+
+        query = posixpath.join(
+            server_setup.DATA_ROOT_DIR,
+            server_setup.BASE_PROFILE_DIR,
+            scenario_info["grid_model"],
+            kind + "_*",
+        )
+        stdin, stdout, stderr = self.data_access.execute_command("ls " + query)
+        if len(stderr.readlines()) != 0:
+            print("No %s profiles available." % kind)
+            version = []
+        else:
+            filename = [os.path.basename(line.rstrip()) for line in stdout.readlines()]
+            version = [f[f.rfind("_") + 1 : -4] for f in filename]
```

Comment:
Do we ever have the case where there is more than one filename on the same line? In this case, does the parsing of the `ls` output still work properly?

Reply:
never thought about it. It seems that on Unix like system `ls` returns values on separate lines, see example below:
```
(PowerSimData) [~/CEM/PowerSimData] (ben/csv) brdo$ ls
ATTRIBUTION.md            Pipfile                   docker-compose.yml        powersimdata.egg-info     setup.py
Dockerfile                Pipfile.lock              optional-requirements.txt pytest.ini                tox.ini
LICENSE                   README.md                 powersimdata              requirements.txt
(PowerSimData) [~/CEM/PowerSimData] (ben/csv) brdo$ ls | wc -l
      14
```

Reply:
Cool, I see the same thing when I'm ssh'ed in, I just wasn't sure whether your example would be parsed as three lines or as fourteen.